expression/scalar_function.go

```diff
@@ -80,7 +80,12 @@ func newFunctionImpl(ctx sessionctx.Context, fold bool, funcName string, retType
 	}
 	fc, ok := funcs[funcName]
 	if !ok {
-		return nil, errFunctionNotExists.GenWithStackByArgs("FUNCTION", funcName)
+		dbFuncName := ctx.GetSessionVars().CurrentDB
+		if dbFuncName != "" {
+			dbFuncName += "."
+		}
```

Comment:
if `dbFuncName == ""`, we should return an `error` to indicate that `ERROR 1046 (3D000): No database selected`.

Reply:
@lizhenda Please take a look at this comment.

Reply:
fix in the latest commit.

Reply:
Could you add one more test case covering`No database selected`

Reply:
Add test case for function without params